include/sensors/core/tasks/capacitive_driver.hpp

```diff
@@ -237,15 +240,24 @@ class FDC1004 {
             stop_continuous_polling(m.id.token);
             return;
         }
+
+        auto raw_capacitance = fdc1004_utils::convert_reads(polling_results[0],
+                                                            polling_results[1]);
+
+        if (data_unstable[int(sensor_id)]) {
+            raw_capacitance = filter.compute(raw_capacitance);
+            data_unstable.set(int(sensor_id), filter.stop_filter());
```

Comment:
Is the logic correct here? My understanding is that `stop_filter` returns `true` if the capdac should stop filtering, aka if the data is stable. That means that, if the data is _not_ stable, `stop_filter` should return false. That will set the bit in `data_unstable` to 0, indicating that the data is stable.

Reply:
Good catch!